### PR TITLE
Promote allocation to v1

### PR DESCRIPTION
**Pull request by roberthbailey**
This is part of #703.

/assign @markmandel
/label kind/breaking

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like you need a `make gen-install` :smile:

**Comment by roberthbailey** [comment]
Yes.... sometimes it's hard to tell which files are edited and which are generated.

**Comment by roberthbailey** [comment]
This is likely to need another helm reset too.

**Comment by roberthbailey** [comment]
The last cloud build failed on helm like the previous PR.

File: pkg/apis/allocation/v1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -18,11 +18,11 @@
...
 // Code generated by deepcopy-gen. DO NOT EDIT.
 
-package v1alpha1
+package v1
 
 import (
-	stable_v1alpha1 "agones.dev/agones/pkg/apis/stable/v1alpha1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha1 "agones.dev/agones/pkg/apis/stable/v1alpha1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (markmandel) [inline pkg/apis/allocation/v1/zz_generated.deepcopy.go]: we've used `metav1` everywhere else (pretty sure that's what k8s uses too as an alias)
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

File: site/content/en/docs/Getting Started/create-fleet.md [hunk 2]
```diff
@@ -176,7 +176,7 @@ For the full details of the YAML file head to the [Fleet Specification Guide]({{
 You should get back a response that looks like the following:
 
 ```yaml
-apiVersion: allocation.agones.dev/v1alpha1
+apiVersion: allocation.agones.dev/v1
```
Comment (markmandel) [inline site/content/en/docs/Getting Started/create-fleet.md]: we'll need to {{< feature >}} this

File: site/content/en/docs/Guides/access-api.md [hunk 3]
```diff
@@ -236,11 +236,11 @@ $ curl http://localhost:8001/apis/stable.agones.dev/v1alpha1/namespaces/default/
 
 # allocate a gameserver from a fleet named 'simple-udp', with GameServerAllocation
 
-$ curl -d '{"apiVersion":"allocation.agones.dev/v1alpha1","kind":"GameServerAllocation","spec":{"required":{"matchLabels":{"stable.agones.dev/fleet":"simple-udp"}}}}' -H "Content-Type: application/json" -X POST http://localhost:8001/apis/allocation.agones.dev/v1alpha1/namespaces/default/gameserverallocations
+$ curl -d '{"apiVersion":"allocation.agones.dev/v1","kind":"GameServerAllocation","spec":{"required":{"matchLabels":{"stable.agones.dev/fleet":"simple-udp"}}}}' -H "Content-Type: application/json" -X POST http://localhost:8001/apis/allocation.agones.dev/v1/namespaces/default/gameserverallocations
```
Comment (markmandel) [inline site/content/en/docs/Guides/access-api.md]: we'll need to {{< feature >}} this

**Comment by markmandel** [comment]
Deleting the helm install, and rerunning!

File: pkg/apis/allocation/v1/zz_generated.deepcopy.go (hunk 1 above)
Comment (markmandel) [reply pkg/apis/allocation/v1/zz_generated.deepcopy.go]: oh wait, this is generated. Ignore this.

Comment (roberthbailey) [reply pkg/apis/allocation/v1/zz_generated.deepcopy.go]: I'm not sure why the generated code decided to change this import name, but I'd rather not fight with it. :)

File: site/content/en/docs/Getting Started/create-fleet.md (hunk 2 above)
Comment (roberthbailey) [reply site/content/en/docs/Getting Started/create-fleet.md]: Good catch. Fixed.

File: site/content/en/docs/Guides/access-api.md (hunk 3 above)
Comment (roberthbailey) [reply site/content/en/docs/Guides/access-api.md]: Good catch. Fixed.

**Comment by roberthbailey** [comment]
I'm seeing a weird error on the e2e tests:

```
Already have image: e2e-runner
cp: cannot stat '/workspace/./build/.gomod/cache/vcs/b9a4b9bbdb4a59723f2348415ad7ffda91568455a1cfd92e97976132bdfbaf57/shallow.lock': No such file or directory
```

This doesn't seem related to my change and I'm not sure how to fix it.

**Comment by markmandel** [comment]
I think Cloud Build is having a sad. Will retry shortly.

**Comment by markmandel** [comment]
Yep,, definite sad.

**Comment by roberthbailey** [comment]
I was having trouble reproducing this failure locally:

```
$ make test-e2e
...
ok  	agones.dev/agones/test/e2e	551.763s
echo "Finishing e2e test runner!"
Finishing e2e test runner!
```

and for some reason after a few re-tries cloud build is now working too.

**Pull request merged**